### PR TITLE
release-23.1: license: Remove the BSL and CCL license files

### PR DESCRIPTION
**Pull request by jlinder**
Backport 1/1 commits from #131793.

/cc @cockroachdb/release

---

With all CockroachDB code now licensed under the CockroachDB Software License (CSL), the BSL and CCL license files, along with the licenses/LICENSE file that references the BSL and CCL licenses, can be removed. This commit removes them.

Part of RE-658

Release note (general change): Change the license cockroach is distributed under to the new CockroachDB Software License (CSL).

---

Release justification: Need to change the cockroach license to the CockroachDB Software License (CSL).

### CODE REVIEW

**Comment by jlinder** [comment]
This will merge after the 23.1 backport for https://github.com/cockroachdb/cockroach/pull/131781 is merged (waiting on #132052  to be merged first).

**Review by rail — APPROVED** [review]

**Review by celiala — APPROVED** [review]

**Pull request merged**
